src/System.Memory/src/System/Buffers/BuffersExtensions.cs

```diff
@@ -54,5 +54,40 @@ public static T[] ToArray<T>(this ReadOnlySequence<T> sequence)
             sequence.CopyTo(array);
             return array;
         }
+
+        /// <summary>
+        /// Writes contents of <paramref name="source"/> to <paramref name="bufferWriter"/>
+        /// </summary>
+        public static void Write<T>(this IBufferWriter<T> bufferWriter, ReadOnlySpan<T> source)
+        {
+            Span<T> destination = bufferWriter.GetSpan();
+
+            // Fast path, try copying to the available memory directly
+            if (source.Length <= destination.Length)
+            {
+                source.CopyTo(destination);
+                bufferWriter.Advance(source.Length);
+                return;
+            }
+
+            while (source.Length > 0)
+            {
+                int writeSize;
+                if (destination.Length == 0)
```

Comment:
Aside from the first time through the loop, this will always be true (since `destination = default`). Consider special casing the first iteration outside the loop.